libevmasm/Assembly.cpp

```diff
@@ -74,6 +75,176 @@ unsigned Assembly::codeSize(unsigned subTagSize) const
+		m_libraries[hash] = _libraryName;
+		return hash;
+	};
+
+	if (srcIndex >= 0 && srcIndex < static_cast<int>(sourceUnitNames().size()))
+		location.sourceName = std::make_shared<std::string>(sourceUnitNames()[static_cast<size_t>(srcIndex)]);
+	else if (srcIndex < -1 || srcIndex >= static_cast<int>(sourceUnitNames().size()))
+		solThrow(AssemblyImportException, "srcIndex out of bound.");
+
+	AssemblyItem result(0);
+
+	if (c_instructions.count(name))
+	{
+		AssemblyItem item{c_instructions.at(name), location};
+		if (!jumpType.empty())
+		{
+			if (item.instruction() == Instruction::JUMP || item.instruction() == Instruction::JUMPI)
+				item.setJumpType(jumpType);
+			else
+				solThrow(
+					AssemblyImportException,
+					"Attribute 'jumpType' set on instruction different from JUMP or JUMPI (was set on instruction '" + name + "')"
+				);
+		}
+		result = item;
+	}
+	else
+	{
+		if (name == "PUSH")
+			result = {AssemblyItemType::Push, u256("0x" + value)};
+		else if (name == "PUSH [ErrorTag]")
+		{
+			solThrowIf(
+				!value.empty(),
+				AssemblyImportException,
+				"Attribute 'value' defined for instruction '" + name + "', but instruction don't need a value."
+			);
+			result = {AssemblyItemType::PushTag, 0};
+		}
+		else if (name == "PUSH [tag]")
+			result = {AssemblyItemType::PushTag, updateUsedTags(u256(value))};
+		else if (name == "PUSH [$]")
+			result = {AssemblyItemType::PushSub, u256("0x" + value)};
+		else if (name == "PUSH #[$]")
+			result = {AssemblyItemType::PushSubSize, u256("0x" + value)};
+		else if (name == "PUSHSIZE")
+		{
+			solThrowIf(
+				!value.empty(),
+				AssemblyImportException,
+				"Attribute 'value' defined for instruction '" + name + "', but instruction don't need a value."
+			);
+			result = {AssemblyItemType::PushProgramSize, 0};
+		}
+		else if (name == "PUSHLIB")
+			result = {AssemblyItemType::PushLibraryAddress, storeLibraryHash(value)};
+		else if (name == "PUSHDEPLOYADDRESS")
+		{
+			solThrowIf(
+				!value.empty(),
+				AssemblyImportException,
+				"Attribute 'value' defined for instruction '" + name + "', but instruction don't need a value."
+			);
+			result = {AssemblyItemType::PushDeployTimeAddress, 0};
+		}
+		else if (name == "PUSHIMMUTABLE")
+			result = {AssemblyItemType::PushImmutable, storeImmutableHash(value)};
+		else if (name == "ASSIGNIMMUTABLE")
+			result = {AssemblyItemType::AssignImmutable, storeImmutableHash(value)};
+		else if (name == "tag")
+			result = {AssemblyItemType::Tag, updateUsedTags(u256(value))};
+		else if (name == "PUSH data")
+			result = {AssemblyItemType::PushData, u256("0x" + value)};
+		else if (name == "VERBATIM")
+		{
+			solThrowIf(
+				!value.empty(),
+				AssemblyImportException,
+				"Attribute 'value' defined for instruction '" + name + "', but instruction don't need a value."
+			);
+			AssemblyItem item(fromHex(value), 0, 0);
+			result = item;
+		}
+		else
+			assertThrow(false, InvalidOpcode, "");
```

Comment:
I think this is validation, not an assert.
```suggestion
			solThrow(InvalidOpcode, "Invalid opcode: " + name);
```